EL-4087 - Marquee - Wizard Custom Step Content

ashley-hunter opened the pull request:
#### Checklist
<!-- Use an 'x' to check those which apply. -->
* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licensed under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue
<!-- Either a Jira URL or a description of the issue that this PR addresses. -->
https://portal.digitalsafe.net/browse/EL-4087

#### Description of Proposed Changes
<!-- Describe what was changed and how it addresses the original issue. -->
- Adding a `context` input on each step to allow them to provide custom data to the step template
- Adding `index` and `context` properties to the step template context
- Improving documentation with new examples and documenting the new context interface and input
- Added test case
- Fixed an incorrect type used by a splitter event

#### Documentation CI URL
<!-- Initiate a build at https://jenkins.swinfra.net/job/SEPG/view/Templates/job/New%20SEPG%20Build/build -->
<!-- Append the branch name to the following URL: -->
https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-4087-Marquee-Wizard-Custom-Content

#### Downstream Build(s)
<!-- Push a branch with the same name in each downstream repository and create a build in Jenkins. -->
<!-- Add the Jenkins build link(s) below: -->
* [caf/ux-aspects-micro-focus](https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~ux-aspects-micro-focus~EL-4087-Marquee-Wizard-Custom-Content~CI/)

buildmachine-sou-jenkins2 commented:
A developer build has not yet been created for this branch. Click [here](https://jenkins.swinfra.net/job/SEPG/job/New%20SEPG%20Build/parambuild/?GITHUB_SERVER=github.com&GITHUB_ORGANISATION=UXAspects&REPO_NAME=UXAspects&BRANCH_NAME=EL-4087-Marquee-Wizard-Custom-Content) to go ahead and create the build...

buildmachine-sou-jenkins2 commented:
CI Build Link:
https://jenkins.swinfra.net/job/SEPG/job/UXAspects/job/UXAspects~UXAspects~EL-4087-Marquee-Wizard-Custom-Content~CI

buildmachine-sou-jenkins2 commented:
The Documentation QA site for this branch has been built:
https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-4087-Marquee-Wizard-Custom-Content

donna-marijne reviewed (changes requested):
Thanks, this looks good. Apart from the badge issue, just a couple of documentation suggestions.

ashley-hunter commented:
I have made the suggested review fixes and also updated the badge directive to accept numbers as inputs rather than solely strings. I have added test cases for this change also.

donna-marijne commented on docs/app/pages/components/components-sections/wizard/marquee-wizard/snippets/step-template.html:
```diff
@@ -0,0 +1,19 @@
+                 let-index="index"
+                 let-context="context">
+
+                    <span class="marquee-wizard-step-title"
```
Indentation issue here.

donna-marijne reviewed (changes requested):
Thank you, great to get the badge issue taken care of here, it will be a much better experience for anyone trying to use it. There's just one indentation issue in the code snippet.

ashley-hunter commented:
Thanks Donna, indentation should now be corrected

donna-marijne reviewed (dismissed):
Pushed a fix. @Sdhilipkumar waiting on your review now.

Sdhilipkumar reviewed (approved):
Looks good.
